frontend/src/actionCreators/mineActionCreator.js

```diff
@@ -9,23 +9,57 @@ import * as API from "@/constants/API";
 import { ENVIRONMENT } from "@/constants/environment";
 import { createRequestHeader } from "@/utils/RequestHeaders";
 
+const submitDisturbances = (type) => ({ data }) => {
+  const disturbanceResponses = type.mine_disturbance_code.map((code) =>
+    axios.post(
+      ENVIRONMENT.apiUrl + API.MINE_TYPES_DETAILS,
+      {
+        mine_type_guid: data.mine_type_guid,
+        mine_disturbance_code: code,
+      },
+      createRequestHeader()
+    )
+  );
+  return Promise.all(disturbanceResponses);
+};
+
+const handleError = (dispatch, reducer) => (err) => {
+  notification.error({
+    message: err.response ? err.response.data.error.message : String.ERROR,
+    duration: 10,
+  });
+  dispatch(error(reducer));
+  dispatch(hideLoading("modal"));
+};
+
+const createMineTypeRequests = (payload, dispatch, reducer) => (response) => {
+  if (payload.mine_types) {
+    const allMineTypes = payload.mine_types.map((type) =>
```

Comment:
I think it might be better yet to pull this out as `submitMineTypes`. Then this will read something like:
```javascript
if (payload.mine_types) {
  const allMineTypes = submitMineTypes
    .then(submitDisturbances(type))
    .catch(handleError(dispatch, reducer));
  return Promise.all(allMineTypes);
}
return response;
```

Which should make it very self-documenting